sdk/tables/Azure.Data.Tables/src/Extensions/TableEntityExtensions.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Azure.Data.Tables
+{
+    internal static class TableEntityExtensions
+    {
+        /// <summary>
+        /// Returns a new Dictionary with the appropriate Odata type annotation for a given propertyName value pair.
+        /// The default case is intentionally unhandled as this means that no type annotation for the specified type is required.
+        /// This is because the type is naturally serialized in a way that the table service can interpret without hints.
+        /// </summary>
+        internal static Dictionary<string, object> ToOdataAnnotatedDictionary<T>(this T entity) where T : TableEntity
+        {
+            var properties = typeof(T).GetProperties(BindingFlags.Instance | BindingFlags.Public);
```

Comment:
Can we reuse logic from DictionaryTableExtensions after the dictionary is created.

Reply:
It's different enough that we'd have to do some gymnastics to factor out the common parts.